NickvisionTubeConverter.Shared/Models/DownloadManager.cs

```diff
@@ -200,9 +201,14 @@ public void AddDownload(Download download, DownloadOptions options)
             _queued.Add(download.Id, (download, options));
             DownloadAdded?.Invoke(this, (download.Id, download.Filename, download.SaveFolder, false));
         }
-        var history = DownloadHistory.Current;
-        history.History[download.MediaUrl] = (Path.GetFileNameWithoutExtension(download.Filename), DateTime.Now, $"{download.SaveFolder}{Path.DirectorySeparatorChar}{download.Filename}");
-        history.Save();
+        var history = (DownloadHistory)Aura.Active.ConfigFiles["downloadHistory"];
+        history.History[download.MediaUrl] = new DownloadHistoryItem
```

Comment:
Using the new constructor this will be
```csharp
history.History[download.MediaUrl] = new DownloadHistoryItem(download.MediaUrl)
{
    Title = Path.GetFileNameWithoutExtension(download.Filename),
    Path = $"{download.SaveFolder}{Path.DirectorySeparatorChar}{download.Filename}"
};
```